mmcai-setup.sh

```diff
@@ -1,62 +1,399 @@
+                * ) create_mysql_secret=false;;
+            esac
+        else
+            create_mysql_secret=true
+        fi
+
+        if $create_mysql_secret; then
+            echo "Enter new billing database secret."
+            MYSQL_ROOT_PASSWORD=''
+            MYSQL_ROOT_PASSWORD_CONFIRMATION='nonempty'
+            until [ "$MYSQL_ROOT_PASSWORD" = "$MYSQL_ROOT_PASSWORD_CONFIRMATION" ]; do
+                read -sp "Billing database root password:" MYSQL_ROOT_PASSWORD
+                echo
+                read -sp "Confirm database root password:" MYSQL_ROOT_PASSWORD_CONFIRMATION
+                echo
+                if [ "$MYSQL_ROOT_PASSWORD" != "$MYSQL_ROOT_PASSWORD_CONFIRMATION" ]; then
+                    log_bad "Passwords do not match."
+                fi
+            done
+        else
+            log "Using existing billing database secret."
+        fi
+    fi
+fi
+
+################################################################################
 
 div
-log "Welcome to MMC.AI setup!"
+echo "COMPONENT: INSTALL"
+echo "MMC.AI Manager:" $install_mmcai_manager
+echo "MMC.AI Cluster:" $install_mmcai_cluster
+echo "NVIDIA GPU Operator:" $install_nvidia_gpu_operator
+echo "Kubeflow:" $install_kubeflow
+echo "cert-manager:" $install_cert_manager
+
 div
+read -p "Confirm selection [y/N]:" confirm_selection
+case $confirm_selection in
+    [Yy]* ) confirm_selection=true;;
+    * ) confirm_selection=false;;
+esac
 
-NAMESPACE="mmcai-system"
+if ! $confirm_selection; then
+    div
+    log_good "Aborting..."
+    exit 0
+fi
 
 div
-log_good "Please provide information for billing database:"
-div
+log_good "Beginning setup..."
 
-read -p "MySQL database node hostname: " mysql_node_hostname
-read -sp "MySQL root password: " MYSQL_ROOT_PASSWORD
-echo ""
+################################################################################
 
+if $install_cert_manager; then
+    div
+    log_good "Installing cert-manager..."
+    helm repo add jetstack https://charts.jetstack.io
+    helm repo update
+    helm install --wait --create-namespace -n cert-manager cert-manager jetstack/cert-manager --version $CERT_MANAGER_VERSION \
+      --set crds.enabled=true
+fi
 
-div
-log_good "Creating namespaces if needed..."
-div
+if $install_kubeflow; then
+    div
+    log_good "Installing Kubeflow..."
 
-## Create namespaces
+    curl -LfsS https://raw.githubusercontent.com/MemVerge/mmc.ai-setup/main/playbooks/sysctl-playbook.yaml | \
+    ansible-playbook -i $ANSIBLE_INVENTORY /dev/stdin
 
-kubectl get namespace $NAMESPACE &>/dev/null || kubectl create namespace $NAMESPACE
-kubectl get namespace mmcloud-operator-system &>/dev/null || kubectl create namespace mmcloud-operator-system
-kubectl get namespace monitoring &>/dev/null || kubectl create namespace monitoring
+    log "Cloning Kubeflow manifests..."
+    git clone https://github.com/kubeflow/manifests.git $TEMP_DIR/kubeflow --branch $KUBEFLOW_VERSION
```

Comment:
Ooh we have to apply our kubeflow configuration patches here.

Here's how deepops does it: https://github.com/NVIDIA/deepops/blob/master/scripts/k8s/deploy_kubeflow.sh#L135-L150

The ones we care about are:

[manifests/common/istio-1-22/istio-install/base/patches/service.yaml ](https://github.com/kubeflow/manifests/blob/v1.9-branch/common/istio-1-22/istio-install/base/patches/service.yaml) -> Change from ClusterIP to NodePort to expose the kubeflow stuff as an external service

[manifests/common/user-namespace/base/params.env](https://github.com/kubeflow/manifests/blob/master/common/user-namespace/base/params.env) -> Change default username to deepops@example.com, and default password to deepops

Then the following `sed`s to use HTTP instead of HTTPS:

```
sed -i 's:JWA_APP_SECURE_COOKIES=true:JWA_APP_SECURE_COOKIES=false:' manifests/apps/jupyter/jupyter-web-app/upstream/base/params.env 

sed -i 's:VWA_APP_SECURE_COOKIES=true:VWA_APP_SECURE_COOKIES=false:' manifests/apps/volumes-web-app/upstream/base/params.env 

sed -i 's:TWA_APP_SECURE_COOKIES=true:TWA_APP_SECURE_COOKIES=false:' manifests/apps/tensorboard/tensorboards-web-app/upstream/base/params.env
```

Reply:
Wouldn't it be better to change mmc.ai-mgmt to also use the default username and password (user@example.com and 12341234) from Kubeflow (even better if we can remove entirely), rather than continue modifying the Kubeflow user to match with DeepOps?

Reply:
I agree, probably that's the better way. But I think the flow of changes would have to be

Zubiao/Dora change the backend -> we change the installation script

If they can do it soon, then sure, we should just keep the default, but otherwise we should add the deepops credentials.

Reply:
I asked about it on one of the group chats, let's see what they say.

Reply:
OK seems kosher, we can merge in the mmc.ai-mgmt PR and then keep the default user@example.com here.